logprep/processor/timestamp_differ/processor.py

```diff
@@ -0,0 +1,66 @@
+"""
+TimestampDiffer
+===============
+
+The `timestamp_differ` can calculate the time difference between two timestamps.
+For further information for the rule language see: :ref:`timestamp_differ_rule`.
+
+Example
+^^^^^^^
+..  code-block:: yaml
+    :linenos:
+
+    - timestampdiffer_name:
+        type: timestamp_differ
+        specific_rules:
+            - tests/testdata/rules/specific/
+        generic_rules:
+            - tests/testdata/rules/generic/
+"""
+from functools import partial, reduce
+
+import arrow
+
+from logprep.abc import Processor
+from logprep.processor.base.exceptions import DuplicationError
+from logprep.processor.timestamp_differ.rule import TimestampDifferRule
+from logprep.util.helper import get_dotted_field_value, add_field_to
+
+
+class TimestampDiffer(Processor):
+    """A processor that calculates the time difference between two timestamps"""
+
+    rule_class = TimestampDifferRule
+
+    def _apply_rules(self, event, rule):
+        source_fields, source_field_timestamp_formats = list(zip(*rule.source_fields))
```

Comment:
here you dissect something in processing, which could be saved separately

Reply:
Was revised in the fix of the other remark.